Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -273,6 +273,42 @@ open class RouteController: NSObject {
         }
         // Check for faster route proactively (if reroutesProactively is enabled)
         refreshAndCheckForFasterRoute(from: location, routeProgress: routeProgress)
+        
+        verifyNavigationTilesVersion()
+    }
+    
+    private var wantsSwitchTilesVersion = false
+    private var wantsRestoreTilesVersion = false
+    
+    private func verifyNavigationTilesVersion() {
+        switch Navigator.shared.tileVersionState {
+        case .shouldFallback(let versions):
+            if !wantsSwitchTilesVersion {
+                wantsSwitchTilesVersion = true
+                delegate?.router(self, wantsSwitchTilesVersion: versions, completion: { [weak self] version in
```

Comment:
It is clear from the code that this completion should be called on the main thread. I would either document this assumption or made sure that the code below is called on the correct thread.